continuous_integration/Dockerfile

```diff
@@ -0,0 +1,70 @@
+ARG BASE_IMAGE=rapidsai/devcontainers:23.06-cpp-cuda11.8-mambaforge-ubuntu22.04
+FROM ${BASE_IMAGE} as stage0
+
+RUN export DEBIAN_FRONTEND=noninteractive       \
+ && apt update                             \
+ && apt install -y --no-install-recommends \
+    coreutils uuid-runtime curl rustc cargo
+
+ENV PYTHONDONTWRITEBYTECODE=1
+ENV SCCACHE_REGION="us-east-2"
+ENV SCCACHE_BUCKET="rapids-sccache-east"
+ENV SCCACHE_S3_KEY_PREFIX=legate-cunumeric-dev
+ENV VAULT_HOST=https://vault.ops.k8s.rapids.ai
+ENV HISTFILE=/home/coder/.cache/._bash_history
+
+ARG REPO_OWNER
+ENV REPO_OWNER=${REPO_OWNER}
+
+ARG REPO_NAME
+ENV REPO_NAME=${REPO_NAME}
+
+ARG GITHUB_TOKEN
+ENV GITHUB_TOKEN=${GITHUB_TOKEN}
```

Comment:
This embeds the GITHUB_TOKEN in the history of the docker image. Please mount secrets into the image instead of define them as build args.

Reply:
Moved this to the build stage and since the final image is created from stage0 this will no longer be visible in the image history: https://github.com/nv-legate/legate.core/actions/runs/5553609192/jobs/10142367105